worlds/sm64ex/__init__.py

```diff
@@ -238,3 +238,19 @@ def extend_hint_information(self, hint_data: typing.Dict[int, typing.Dict[int, s
                     for location in region.locations:
                         er_hint_data[location.address] = entrance_name
             hint_data[self.player] = er_hint_data
+
+    def write_spoiler(self, spoiler_handle: typing.TextIO) -> None:
+        # Write calculated star costs to spoiler.
+        star_cost_spoiler_header = '\n\n' + self.player_name + ' Star Costs for Super Mario 64:\n\n'
+        spoiler_handle.write(star_cost_spoiler_header)
+        # - Reformat star costs dictionary in spoiler to be a bit more readable.
+        star_costs_spoiler = self.star_costs.copy()
+        star_costs_spoiler['First Floor Big Star Door'] = star_costs_spoiler.pop('FirstBowserDoorCost')
+        star_costs_spoiler['Basement Big Star Door'] = star_costs_spoiler.pop('BasementDoorCost')
+        star_costs_spoiler['Second Floor Big Star Door'] = star_costs_spoiler.pop('SecondFloorDoorCost')
+        star_costs_spoiler['MIPS 1'] = star_costs_spoiler.pop('MIPS1Cost')
+        star_costs_spoiler['MIPS 2'] = star_costs_spoiler.pop('MIPS2Cost')
+        star_costs_spoiler['Endless Stairs'] = star_costs_spoiler.pop('StarsToFinish')
+        star_cost_key_maxlen = len(max(star_costs_spoiler, key=len))
+        for star, cost in star_costs_spoiler.items():
+            spoiler_handle.write(f"{star:{star_cost_key_maxlen}s} = {cost}\n")
```

Comment:
Popping by name is slow. Also `star_cost_key_maxlen` is a constant. At the very least it doesn't have to be recalculated for each world, you can either use an actual raw number, like in this suggestion, or add a constant outside of this function
```suggestion
        star_costs_spoiler = {}
        star_costs_copy = self.star_costs.copy()
        star_costs_spoiler['First Floor Big Star Door'] = star_costs_copy['FirstBowserDoorCost']
        star_costs_spoiler['Basement Big Star Door'] = star_costs_copy['BasementDoorCost']
        star_costs_spoiler['Second Floor Big Star Door'] = star_costs_copy['SecondFloorDoorCost']
        star_costs_spoiler['MIPS 1'] = star_costs_copy['MIPS1Cost']
        star_costs_spoiler['MIPS 2'] = star_costs_copy['MIPS2Cost']
        star_costs_spoiler['Endless Stairs'] = star_costs_copy['StarsToFinish']
        for star, cost in star_costs_spoiler.items():
            spoiler_handle.write(f"{star:{26}s} = {cost}\n")
```

Reply:
Keeping a constant seems quite unwieldy to maintain, I'd like to avoid that if possible.

Reply:
You can do `star_cost_key_maxlen = len(max(['First Floor Big Star Door', 'Basement Big Star Door', 'Second Floor Big Star Door', 'MIPS 1', 'MIPS 2', 'Endless Stairs]], key=len))` outside of the function then.